usefulUtilities/sit/sitClient.js

```diff
@@ -233,6 +236,110 @@
         }
     }
 
+
+    function maybeClearRequestSitDataTimer() {
+        if (requestSitDataTimer) {
+            Script.clearTimeout(requestSitDataTimer);
+            requestSitDataTimer = false;
+        }
+    }
+
+
+    var requestSitDataTimer = false;
+    var REQUEST_SIT_DATA_TIMEOUT_MS = 12000;
+    var MAX_REQUEST_DISTANCE_M = 40;
+    function requestSitDataTimerCallback() {
+        if (DEBUG) {
+            console.log("sitClient.js: We're inside `requestSitDataTimerCallback()` for Entity ID: " + _this.entityID + "...");
+        }
+
+        requestSitDataTimer = false;
+
+        var entityPosition = Entities.getEntityProperties(_this.entityID, ["position"]).position;
+        var distance = Vec3.distance(entityPosition, Camera.position);
+        if (distance <= MAX_REQUEST_DISTANCE_M) {
+            if (DEBUG) {
+                console.log("sitClient.js: The camera is close enough to this seat! Requesting sit data from sitServer...");
+            }
+            requestSitData(_this.entityID);
+        } else {
+            if (DEBUG) {
+                console.log("sitClient.js: The camera is too far away from this seat! Waiting " + REQUEST_SIT_DATA_TIMEOUT_MS + "ms before trying again...");
+            }
+            requestSitDataTimer = Script.setTimeout(requestSitDataTimerCallback, REQUEST_SIT_DATA_TIMEOUT_MS);
+        }
+    }
+
+
+    function maybeClearRequestSitDataTimeoutTimer() {
+        currentRequestAttempts = 0;
+        if (requestSitDataTimeoutTimer) {
+            Script.clearTimeout(requestSitDataTimeoutTimer);
+            requestSitDataTimeoutTimer = false;
+        }
+    }
+
+
+    var currentRequestAttempts = 0;
+    var MAX_REQUEST_ATTEMPTS = 3;
+    function requestSitDataTimeoutTimerCallback() {
+        requestSitDataTimeoutTimer = false;
+        currentRequestAttempts++;
+        if (currentRequestAttempts <= MAX_REQUEST_ATTEMPTS) {
+            console.log("sitClient.js: Request for sit data timed out for Entity ID: " + _this.entityID +
+                ". Trying again. This will be attempt " + currentRequestAttempts + "/" + MAX_REQUEST_ATTEMPTS);
+            requestSitData(_this.entityID);
+        } else {
+            currentRequestAttempts = 0;
+            console.log("sitClient.js: Request for sit data timed out for Entity ID: " + _this.entityID +
+                ". " + MAX_REQUEST_ATTEMPTS + " requests for data failed. Not trying again.");
+        }
+    }
+
+
+    var requestSitDataTimeoutTimer = false;
+    var REQUEST_SIT_DATA_TIMEOUT_MS = 6000;
+    function requestSitData(id) {
+        if (DEBUG) {
+            console.log("sitClient.js: Requesting sit data...");
+        }
+
+        currentRequestAttempts++;
```

Comment:
Was this supposed to be incremented twice for every timeout? (287 and 307)

Reply:
Sorry, I don't follow your question. Can you please rephrase?

Reply:
'requestSitDataTimeoutTimerCallback' adds one to 'currentRequestAttempts' at line 287 and then calls 'requestSitData' which adds one to 'currentRequestAttempts' again at line 307.

Reply:
ooooh, great catch! this logic is wrong! fixing.